src/Lean/Widget/InteractiveCode.lean

```diff
@@ -15,6 +15,36 @@ with environment and subexpression information. -/
 namespace Lean.Widget
 open Server
 
+inductive HighlightColor where
+  | green
+  | blue
+  | red
+  | yellow
+  | purple
+
+def HighlightColor.toString : HighlightColor → String
+  | .green  => "green"
+  | .blue   => "blue"
+  | .red    => "red"
+  | .yellow => "yellow"
+  | .purple => "purple"
+
+def HighlightColor.fromString : String → Except String HighlightColor
+  | "green"  => Except.ok HighlightColor.green
+  | "blue"   => Except.ok HighlightColor.blue
+  | "red"    => Except.ok HighlightColor.red
+  | "yellow" => Except.ok HighlightColor.yellow
+  | "purple" => Except.ok HighlightColor.purple
+  | s        => Except.error s!"expected an HighlightColor ctor string but got {s}"
+
+instance : ToString HighlightColor := ⟨HighlightColor.toString⟩
+
+instance : FromJson HighlightColor where
```

Comment:
I think `deriving FromJson, ToJson` should have the same effect, although you'd still need the manual `HighlightColor.toString`.

Reply:
ah I didn't realise deriving worked on enums now :)

Reply:
I've just deleted the toJson fromJson becuase it's not sent to the client anyway

Reply:
We still have the manual instances for `HighlightColor`

Reply:
oops